lib/Echidna/Types/Tx.hs

```diff
@@ -19,6 +19,7 @@ import Echidna.Types.Signature (SolCall)
 data TxCall = SolCreate   ByteString
             | SolCall     SolCall
             | SolCalldata ByteString
+            | NoCall
```

Comment:
I get the semantics from the context of this PR but I think this data constructor can be confusing in future. Can we document the semantics here? 

Looking further, as this is not really a call but rather other event altering the VM state maybe we could refactor that into something like:
```haskell
data TemporalEvent = AddDelay
data EventType = TxCall TxCall | TemporalEvent TemporalEvent
```
This would also eliminate the need of `_              -> error "NoCall"` in `setupTx`. This could involve a bigger refactor to make space for future expansions as `setupTx` and `Tx` type don't handle just Tx calls anymore.

Reply:
Yes, that's a reasonable idea, but yes, it will require a larger refactor, for such a small feature. This should be the only "special" event